### DISCUSSION TITLE
Accessing the 3rd serial port

### DISCUSSION THREAD

**Discussion by bench2012** [arendst/Tasmota#18542] [Q&A] [answered]
Hi,

I figure out how to use the TCP serial bridge, but it was only for 2nd UART (not console). I believe the ESP32 have a 3rd UART, is there any way to add another serial bridge for that?

Many thanks,

Bench

**Comment by barbudor** [comment, marked as answer]
It depends what you want to achieve
Uart0 is always for the console
Uart1 and Uart2 can be used by drivers including TCPbridge or others
But most of Tasmota driver can only be instanciated once
So you can't have 2 TCPbridge although you can have 1 TCPbridge and 1 SerialBridge

**Reply by sfromis** [reply]
When it is ESP32, you could also use Berry to do some programming for the serial processing, possibly a viable strategy instead of bridging serial.

**Comment by bench2012** [comment]
Hi,

Thanks for the reply. I was thinking of using tasmota as a cluster power switch and console monitor. I have written some basic web interface for relay control with TCPserial bridge for the three serial port on esp32 and wonder if tasmota can do a better job with more features. 

No worries.

Thanks anyway.
